### PR TITLE
bootstrap_timeline: rename initdb_path to pgdata_path

### PR DESCRIPTION
**Pull request by arpad-m**
This is a rename without functional changes, in preparation for #5912.

Split off from #5912 as per review request.

### CODE REVIEW

File: pageserver/src/tenant.rs [hunk 1]
```diff
@@ -2921,26 +2921,25 @@ impl Tenant {
 
         // an uninit mark was placed before, nothing else can access this timeline files
         // current initdb was not run yet, so remove whatever was left from the previous runs
-        if initdb_path.exists() {
-            fs::remove_dir_all(&initdb_path).with_context(|| {
-                format!("Failed to remove already existing initdb directory: {initdb_path}")
+        if pgdata_path.exists() {
+            fs::remove_dir_all(&pgdata_path).with_context(|| {
+                format!("Failed to remove already existing initdb directory: {pgdata_path}")
```
[the remark this answers is not in the document]
Comment (problame) [reply pageserver/src/tenant.rs:2926]: fine to leave as is; callers add proper context

**Review by problame — APPROVED** [review]

**Review by koivunej — APPROVED** [review]
Unsure why a local variable rename needed to be split off. This does not change the remote name so all good.

**Comment by arpad-m** [comment]
I'm getting test failures with:

```
2023-11-27T16:50:58.9845784Z [gw0] linux -- Python 3.9.2 /github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/bin/python
2023-11-27T16:50:58.9846661Z test_runner/regress/test_download_extensions.py:144: in test_remote_extensions
2023-11-27T16:50:58.9847274Z     cur.execute("CREATE EXTENSION anon")
2023-11-27T16:50:58.9848242Z E   psycopg2.errors.UndefinedFile: could not open directory "/tmp/neon/pg_install/v14/share/postgresql/extension/anon": No such file or directory
2023-11-27T16:50:58.9849072Z 
2023-11-27T16:50:58.9849343Z During handling of the above exception, another exception occurred:
2023-11-27T16:50:58.9850064Z test_runner/regress/test_download_extensions.py:148: in test_remote_extensions
2023-11-27T16:50:58.9851034Z     assert "pgcrypto" in str(err), f"unexpected error creating anon extension {err}"
2023-11-27T16:50:58.9852276Z E   AssertionError: unexpected error creating anon extension could not open directory "/tmp/neon/pg_install/v14/share/postgresql/extension/anon": No such file or directory
2023-11-27T16:50:58.9853008Z E     
2023-11-27T16:50:58.9853698Z E   assert 'pgcrypto' in 'could not open directory "/tmp/neon/pg_install/v14/share/postgresql/extension/anon": No such file or directory\n'
2023-11-27T16:50:58.9855253Z E    +  where 'could not open directory "/tmp/neon/pg_install/v14/share/postgresql/extension/anon": No such file or directory\n' = str(UndefinedFile('could not open directory "/tmp/neon/pg_install/v14/share/postgresql/extension/anon": No such file or directory\n'))
```

**Comment by arpad-m** [comment]
The failures are on all postgres pre-16 regression test runners, and also happened in earlier CI runs, so they reproduce reliably. I suppose at this point it's not this PR's fault.

**Comment by arpad-m** [comment]
Slack discussion for the test failure: https://neondb.slack.com/archives/C059ZC138NR/p1701094775139019

**Pull request merged**